pkg/util/util.go

```diff
@@ -378,46 +378,27 @@ func getIPFromInterface(intfName string, nw networkInterfacer) (net.IP, error) {
 	return nil, nil
 }
 
-func flagsSet(flags net.Flags, test net.Flags) bool {
-	return flags&test != 0
-}
-
-func flagsClear(flags net.Flags, test net.Flags) bool {
-	return flags&test == 0
-}
-
-func chooseHostInterfaceNativeGo() (net.IP, error) {
-	intfs, err := net.Interfaces()
+//chooseHostInterfaceNativeGo is a method used to fetch an IP for a daemon.
+//It iterates over the interfaces returned by Interfaces() method
+//and picks the first interface which is not Loopback.
+//This method may pick docker or other bridges
+func chooseHostInterfaceNativeGo(nw networkInterfacer) (net.IP, error) {
```

Comment:
127.0.0.1 will not work for actual use.  It might be enough for tests to pass.  Let me run some manual tests tomorrow.

@quinton-hoole who is also picking up a bunch of networking related topics (FYI, no action required yet :)